lms/djangoapps/experiments/flags.py

```diff
@@ -158,17 +181,17 @@ def get_bucket(self, course_key=None, track=True):
                 break
 
         session_key = 'tracked.{}'.format(experiment_name)
-        if track and hasattr(request, 'session') and session_key not in request.session:
+        if track and hasattr(request, 'session') and session_key not in request.session and user == request.user:
             segment.track(
-                user_id=request.user.id,
+                user_id=user.id,
                 event_name='edx.bi.experiment.user.bucketed',
                 properties={
                     'site': request.site.domain,
                     'app_label': self.waffle_namespace.name,
                     'experiment': self.flag_name,
                     'course_id': str(course_key) if course_key else None,
                     'bucket': bucket,
-                    'is_staff': request.user.is_staff,
+                    'is_staff': user.is_staff,
```

Comment:
We *could* check whether you're masquerading as yourself (staff) or a generic learner here. But maybe we always want to send values for your real self.

Reply:
I think we don't want to send this event at all if you're masquerading.